### PR TITLE
Replace strlen with strnlen in native code

### LINKED ISSUE #4975: Remove use for strlen from native code
Remove and or replace the use for strlen from native code

### PR DESCRIPTION
**Pull request by TravisEz13**
fixes #4975 
fix expected vs actual for easier troubleshooting

### CODE REVIEW

File: src/libpsl-native/test/test-createsymlink.cpp [hunk 1]
```diff
@@ -92,7 +92,7 @@ TEST_F(CreateSymLinkTest, SymLinkToFile)
     std::string target = FollowSymLink(fileSymLink.c_str());
     char buffer[PATH_MAX];
     std::string expected = realpath(file, buffer);
-    EXPECT_EQ(target, expected);
+    EXPECT_EQ(expected, target);
```
Comment (daxian-dbw) [inline src/libpsl-native/test/test-createsymlink.cpp]: Can you please explain a bit about why you made the changes to `EXPECT_EQ`?

**Comment by daxian-dbw** [comment]
Again, I think we agreed to submit PR targeting master branch, and then cherrypick commits to 6.0.0. Is that process changed?

File: src/libpsl-native/test/test-createsymlink.cpp (hunk 1 above)
Comment (TravisEz13) [reply src/libpsl-native/test/test-createsymlink.cpp]: The pattern for `EXPECT_EQ` is `EXPECT_EQ(<expected>,<actual>)` when you call it with the parameters reverse it makes the error very confusing.

**Review by daxian-dbw — APPROVED** [review]

**Comment by daxian-dbw** [comment]
Note that the `libpsl` nuget package needs to be updated once this change is merged.

**Pull request merged**